devops/versioning/default.rst

```diff
@@ -0,0 +1,70 @@
+  just have a new revision of that version.
+- **Patch**: Increasing the **patch** version of a package means that only internal changes were done, in practice it means change to files
+  that are not public headers of the package. This "patch" version can avoid having to re-build consumers of this package, for
+  example if the current package getting a new "patch" version is a static library, all other packages that implement static
+  libraries that depend on this one do not need to be re-built from source, as depending on the same public interface headers
+  guarantee the same binary.
+- **Minor**: If changes are done to package public headers, in an API source compatible way, then the recommendation would be to increase
+  the **minor** verson of a package. That means that other packages that depend on it will be able to compile without issues, 
+  but as there were modification in public headers (that could contain C++ templates or other things that could be inlined in
+  the consumer packages), then those consumer packages need to be rebuilt from source to incorporate these changes.
+- **Major**: If API breaking changes are done to the package public headers, then increasing the **major** version is recommended. As the
+  most common recommended version-range is something like ``dependency/[>1.0 <2]``, where the next major is excluded, that means
+  that publishing these new versions will not break existing consumers, because they will not be used at all by those consumers,
+  because their version ranges will exclude them. It will be necessary to modify the consumers recipes and source code (to fix
+  the API breaking changes) to be able to use the new major version.
+
+
+Note that while this is close to the standard "semver" definition of version and version ranges, the C/C++ compilation model
+needs to introduce a new side effect, that of "needing to rebuild the consumers", following the logic explained above in the
+``embed`` and ``non_embed`` cases.
+
+
+This is just the default recommended versioning approach, but Conan allows to change these defaults, as it implements an extension of the "semver" standard that allows any number of digits,
+letters, etc, and it also allows to change the ``package_id`` modes to define how different versions of the dependencies affect
+the consumers binaries. See :ref:`how to customize the dependencies package_id modes<reference_binary_model_custom_compatibility_dependencies>`.
+
+
+.. note::
+
+    **Best practices**
+
+    - It is not recommended to use other package reference fields, as the ``user`` and ``channel`` to represent changes in the source code,
+      or other information like the git branch, as this becomes "viral" requiring changes in the ``requires`` of the consumers. Furthermore,
+      they don't implement any logic in the build model with respect to which consumers need to be rebuilt.
```

Comment:
A good idea would be to provide some insight for where `user` and `channel` _can_ be used. After all these years, I still believe override is a useful dev tool which matches nicely with custom user/channel which helps making dev packages public without exposing dev remote.

Reply:
The ``user/channel`` and other versioning, CI triggering and branch merging, and similar strategies will follow in a later section of the tutorial, at this stage I think it is better to keep it as simple as possible and focus on the pure integration and build.